src/main/java/com/spotify/reaper/cassandra/JmxProxy.java

```diff
@@ -188,8 +188,13 @@ public BigInteger apply(String s) {
   @NotNull
   public List<String> tokenRangeToEndpoint(String keyspace, RingRange tokenRange) {
     checkNotNull(ssProxy, "Looks like the proxy is not connected");
-    Set<Map.Entry<List<String>, List<String>>> entries =
-        ssProxy.getRangeToEndpointMap(keyspace).entrySet();
+    Set<Map.Entry<List<String>, List<String>>> entries;
+    try {
+       entries = ssProxy.getRangeToEndpointMap(keyspace).entrySet();
+    } catch (Exception e) {
```

Comment:
Rather than catching this exception here, I think we should let it throw and catch it in RepairRunner::repairSegment. When caught there, rather than going to ERROR state (which happens in the if clause right after), just do a context.repairManager.scheduleRetry(this) and return. That could fix pl4's problem, and would be much better than resuming ERROR, because ERROR should mean unrecoverable and should ideally only occur in truly unrecoverable situations.